hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -423,7 +423,8 @@ private[hudi] object HoodieSparkSqlWriter {
     hiveSyncConfig.verifyMetadataFileListing = parameters(HoodieMetadataConfig.METADATA_VALIDATE_PROP).toBoolean
     hiveSyncConfig.ignoreExceptions = parameters.get(HIVE_IGNORE_EXCEPTIONS_OPT_KEY).exists(r => r.toBoolean)
     hiveSyncConfig.supportTimestamp = parameters.get(HIVE_SUPPORT_TIMESTAMP).exists(r => r.toBoolean)
-    hiveSyncConfig.autoCreateDatabase = parameters.get(HIVE_AUTO_CREATE_DATABASE_OPT_KEY).exists(r => r.toBoolean)
+    hiveSyncConfig.autoCreateDatabase = parameters.getOrElse(HIVE_AUTO_CREATE_DATABASE_OPT_KEY,
```

Comment:
this is not strictly required. hiveSyncConfig.autoCreateDatabase has a default value to true which is DEFAULT_HIVE_AUTO_CREATE_DATABASE_OPT_KEY.

Reply:
yeh I agree, not sure why would it set to false then. We can close the PR if this is not required

Reply:
I think it's necessary to set it to false in this way, which goes against the default

Reply:
my bad. thanks for clarifying.